### PR TITLE
KIALI-1217 Add a new Workload details page

### PR DESCRIPTION
**Pull request by aljesusg**
** Describe the change **

Add the page for WorkloadDetails
workload_details

** Issue reference **

[KIALI-12-17](https://issues.jboss.org/browse/KIALI-1217)

### CODE REVIEW

**Comment by aljesusg** [comment]
Hooray !!! Updated

**Comment by lucasponce** [comment]
@aljesusg thanks !
One comment, I think we are going to have "Health" at Workload level, similar as it was in Metrics.
So I suggest to create the icon and perhaps use the replica info as part of the tooltip.
We can implement it in a following step.

**Comment by lucasponce** [comment]
Another comment.
In the current breadcrumb style (this will be migrated in the future but meanwhile we should use the current one).
We should have a "Workload Info" like in Services:
image
As we will change add in a next step the "Workload Inbound Metrics" / "Workload Outbound Metrics" tabs.

File: src/types/IstioObjects.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import Namespace from './Namespace';
```
Comment (lucasponce) [inline src/types/IstioObjects.ts]: The file IstioObjects.ts is confusing as it is a little bit ambiguous, in theory the pattern for naming was to use the elements close of the pages.
I understand the goal, it was to reuse some common types between workloads/services.
But well, I guess we can refactor in the future, I don't have a better name.
Once all main entities are consolidated perhaps we can pure refactor some interfaces and classes and clean something as I am sure we have some redundant classes under the types folder.
But no any action required, just leaving here a note to not forget.

File: src/types/Workload.ts [hunk 2]
```diff
@@ -1,8 +1,36 @@
...
+  labels: { [key: string]: string[] };
+  createdAt: Date;
+  resourceVersion: string;
+  replicas: Number;
+  availableReplicas: Number;
+  unavailableReplicas: Number;
+  autoscaler: Autoscaler;
+}
+
 export const WorloadLink = (ns: string, name: string) => {
```
Comment (lucasponce) [inline src/types/Workload.ts]: remember, methods start with lowercase, I think I saw more in the code.

**Comment by lucasponce** [comment]
Another topic I see:
image
The tooltip with number of warnings is good but this requires that you would enter in some details to show the warning details.
As we don't plan to have it, I would use the same way health did: showing the warning/error details in the tooltip, if not that information is missing and user doesn't know what is wrong in the pod.

Also, it happens as well as workload level, there is no warnings related the label validation, and that information should appear at this level as well.

**Comment by aljesusg** [comment]
Comments addresed @lucasponce . Thanks
Added IstioSidecarIcon if it's enabled
latest

How we are displaying now the errors in pods
pods_warning

**Comment by aljesusg** [comment]
increased the width in the tooltip
pods_warning

**Comment by aljesusg** [comment]
hooray !!! Thanks @lucasponce

**Pull request merged**